### ISSUE TITLE
Design: Changes to the date selector for queries, for ease of use and physics 

### ISSUE THREAD

**Issue by joethreepwood** [PostHog/posthog#9453]
## Is your feature request related to a problem?

This is a design nit I noticed, so tagging @clarkus 

Currently if you select a date range option on a query you're building you see this: 

<img width="442" alt="Screenshot 2022-04-19 at 13 40 36" src="https://user-images.githubusercontent.com/84011561/164006191-721b1be4-ad83-4d4a-bab6-95c82ac1c048.png">

And then you have to select 'Start Date' or 'End date' and then it expands to this:

<img width="715" alt="Screenshot 2022-04-19 at 13 38 03" src="https://user-images.githubusercontent.com/84011561/164006285-90199a0e-199c-42d3-a77e-366e4fab90fc.png">

The problems I see are: 

1. This adds an extra click to the flow, because you have to select 'Start date' only to probably then click off of it and choose a date from the calendar view. 
2. May hasn't happened yet. It's non-sensical to let me choose a date which hasn't happened yet, either as a start date or an end date. We seem to currently default to showing this month and next month. 

## Describe the solution you'd like

1. Would seem easier if the calendar view appeared by default and the first click there defaulted to the start date. 
2. Don't default to showing dates in future months on the calendar view. 

This will mean less clicks and better inputs. Faster, better insights! Woo!

## Describe alternatives you've considered

None because my first ideas are always my best ideas.

**Comment by clarkus** [comment]
I think this is mostly an issue of setting reasonable defaults. I agree that we shouldn't allow for future date selection. Secondary to that, we could trigger selection states automatically for some scenarios. It's problematic to click the date control, have to click the start date control, and then actually pick a date. We could just immediately show the date selection prompt here. We could also default the end date to the current date. This would make it so that we're always picking a valid time range and remove the need to select a second date for a common use case. 

I recently redesigned the date picker because we were considering absolute date support for cohort creation. It was removed from scope, but the work is still available at https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=2317%3A2533. I'm mentioning because this iteration overcomes the multi-part range selection by including the range control in the picker itself. Generally a user will just be picking a date or two dates to form a range. If they need to quick edit the values, they can do that via the date controls in the bottom of the component. 

![calendar pickers](https://user-images.githubusercontent.com/254612/164036464-26b2a5bf-b2c1-424d-9190-611dd4e35baa.png)
